partner_compassion/models/partner_compassion.py

```diff
@@ -533,6 +535,26 @@ def compute_geopoint(self):
             partner.advocate_details_id.write(vals)
         return True
 
+    def update_state_from_zip(self, vals):
+        zip_ = vals.get("zip")
+        country_id = vals.get("country_id") or self.country_id.id
```

Comment:
self could be a recordset.
```suggestion
        country_id = vals.get("country_id") or self[:1].country_id.id
```